src/core/tsi/ssl_transport_security.cc

```diff
@@ -953,6 +964,36 @@ static int RootCertExtractCallback(int preverify_ok, X509_STORE_CTX* ctx) {
   return preverify_ok;
 }
 
+// X509_STORE_set_get_crl() sets the function to get the crl for a given
+// certificate x. When found, the crl must be assigned to *crl. This function
+// must return 0 on failure and 1 on success. If no function to get the issuer
+// is provided, the internal default function will be used instead.
+static int GetCrlFromProvider(X509_STORE_CTX* ctx, X509_CRL** crl_out,
+                              X509* x) {
+  SSL* ssl = static_cast<SSL*>(
+      X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx()));
+  SSL_CTX* ssl_ctx = SSL_get_SSL_CTX(ssl);
+  grpc_core::experimental::CrlProvider* provider =
+      static_cast<grpc_core::experimental::CrlProvider*>(
+          SSL_CTX_get_ex_data(ssl_ctx, g_ssl_ctx_ex_crl_provider_index));
+
+  char* buf = X509_NAME_oneline(X509_get_issuer_name(x), nullptr, 0);
```

Comment:
Can you explain what this line is doing? We're just extracting the issuer and converting it to a string?

Reply:
Exactly - `X509_get_issuer_name` returns an `X509_NAME` type object, which `X509_NAME_oneline` converts into a `char*` buffer. `X509_NAME_oneline` can take an input buffer and length and put the data in there, or if that input is null it allocates a new buffer and returns it.